glean-core/src/upload/mod.rs

```diff
@@ -98,22 +98,32 @@ impl PingUploadManager {
     }
 
     /// Creates a `PingRequest` and adds it to the queue.
-    pub fn enqueue_ping(&self, uuid: &str, url: &str, body: JsonValue) {
+    pub fn enqueue_ping(&self, uuid: &str, path: &str, body: JsonValue) {
         let mut queue = self
             .queue
             .write()
             .expect("Can't write to pending pings queue.");
-        let request = PingRequest::new(uuid, url, body);
+        let request = PingRequest::new(uuid, path, body);
         queue.push_back(request);
     }
 
-    /// Clears the pending pings queue.
-    pub fn clear_ping_queue(&self) {
+    /// Clears the pending pings queue, leaves the deletion-request pings.
+    pub fn clear_ping_queue(&self) -> RwLockWriteGuard<'_, VecDeque<PingRequest>> {
         let mut queue = self
             .queue
             .write()
             .expect("Can't write to pending pings queue.");
+
+        let mut deletion_requests = Vec::new();
+        for ping in queue.iter() {
```

Comment:
This whole loop can be simplified to:

```rust
queue.retain(|ping| ping.is_deletion_request());
```

See [`retain`](https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.retain)